Desktop/ReactNativeNavigator-start/W10/FavoritesScreen.tsx

```diff
@@ -0,0 +1,281 @@
+      </View>
+    );
+  }
+
+  if (!animeData) {
+    // Обрабатываем случай, если данные не были получены корректно
+    return (
+      <View style={styles.errorContainer}>
+        <Text style={styles.errorText}>Error fetching anime data</Text>
+      </View>
+    );
+  }
+
+  return (
+    <TouchableOpacity
+      style={styles.favoriteItemContainer}
+      onPress={() => navigation.navigate("AnimeDetail", { animeUrl: url })}
+    >
+      <Image
+        source={{ uri: animeData.images.jpg.image_url }}
+        style={styles.favoriteItemImage}
+      />
+      <View style={styles.favoriteItemInfo}>
+        <Text style={styles.favoriteItemNameText}>{animeData.title}</Text>
+        <Text style={styles.favoriteItemRating}>
+          Rating: {animeData.score.toFixed(2)}
+        </Text>
+      </View>
+      <TouchableOpacity
+        style={styles.removeButton}
+        onPress={() => onRemove(url)}
+      >
+        <Ionicons name="trash-bin" size={20} color="white" />
+      </TouchableOpacity>
+    </TouchableOpacity>
+  );
+};
+
+type FavoritesScreenNavigationProp = StackNavigationProp<
+  YourStackParamList,
+  "FavoritesScreen"
+>;
+
+const FavoritesScreen: React.FC<{
+  navigation: FavoritesScreenNavigationProp;
+}> = ({ navigation }) => {
+  const [favorites, setFavorites] = useState<string[]>([]);
+  const [searchQuery, setSearchQuery] = useState("");
+  const [filteredFavorites, setFilteredFavorites] = useState<string[]>([]);
+  const animeDataMap: Record<string, AnimeData> = {};
+  const [isLoading, setIsLoading] = useState(true);
+
+  useFocusEffect(
+    useCallback(() => {
+      const fetchFavorites = async () => {
+        try {
+          const keys = await AsyncStorage.getAllKeys();
+          const data = await AsyncStorage.multiGet(keys);
+          const favoriteAnime = data
+            .filter(([, value]) => value === "true")
+            .map(([key]) => key);
+
+          // Используем Promise.all для асинхронной загрузки всех данных
+          const animeDataPromises = favoriteAnime.map(async (url) => {
+            try {
+              const response = await fetch(url);
+              if (!response.ok) {
+                throw new Error("Network response was not ok");
+              }
+              const animeData = await response.json();
+              animeDataMap[url] = animeData.data;
+            } catch (error) {
+              console.error("Error fetching anime data:", error);
+              // Если произошла ошибка при загрузке данных, пропустить и продолжить
+            }
+          });
+
+          await Promise.all(animeDataPromises);
+
+          setFavorites(favoriteAnime);
+          setIsLoading(false);
+        } catch (error) {
+          console.error("Error fetching favorite anime:", error);
+        }
+      };
+      fetchFavorites();
+    }, [])
+  );
+
+  useEffect(() => {
+    const filterFavorites = () => {
+      const filtered = favorites.filter((title) =>
```

Comment:
The filtering by name doesn't work because you are only searching the url. You can console.log() the results of `filtered`  You are treating favorites as an array of string instead of an array of AnimeData on line     setSearchQuery(text);